src/blop/integrations/ax/agent.py

```diff
@@ -219,6 +222,34 @@ def _unpack_parameters(self, parameterizations: list[TParameterization]) -> list
 
         return unpacked_list
 
+    def convert_to_dictonary(self, db) -> dict[str, list[Any]]:
+        """
+        Converts the data that arrives from either databroker as a pd.Dataframe or through
+        tiled as a xr.DataArray into a dictonary
+
+        Parameters
+        ----------
+        db : databroker or tiled
+            The databroker or tiled instance
+        """
+        dictonary = {}
+        if isinstance(db, tiled.client.container.Container):
+            tiled_data = db["streams", "primary"].read()
+            for var_name, data_array in tiled_data.data_vars.items():
+                dictonary[var_name] = data_array.values.flatten().tolist()
+            tiled_image = db["streams", "primary", "bl_det_image"].read().astype(float)
+            dictonary["bl_det_image"] = tiled_image
+            return dictonary
+
+        elif isinstance(db, databroker.v1.Header):
+            data = db.table(fill=True)
+            for i in data:
+                dictonary[i] = data[i].to_list()
+            return dictonary
+
+        else:
+            raise ValueError("Unknown data source.")
```

Comment:
Remove this in favor of `self.data_access` in the same pattern you use in the other Agent. This code already exists in `data_access.py` so let's re-use in this agent class as well.